src/templates/singleResource.js

```diff
@@ -0,0 +1,256 @@
+    const ytIndex = urlArr.findIndex(e => e.includes("youtu"))
+    const alternateBaseUrlIndex = urlArr.findIndex(e => e.includes("youtu.be"))
+    const watchModeIndex = urlArr.findIndex(e => e.includes("watch"))
+    const embedModeIndex = urlArr.findIndex(e => e.includes("embed"))
+
+    // URL is format youtu.be, we need to update that
+    if (alternateBaseUrlIndex >= 0) {
+      urlArr.splice(ytIndex, 1, "youtube.com")
+    }
+
+    // URL has /watch we need to remove that
+    if (watchModeIndex >= 0) {
+      urlArr.splice(watchModeIndex, 1)
+    }
+
+    if (ytIndex >= 0 && embedModeIndex === -1) {
+      // We need to add /embed
+      urlArr.splice(ytIndex + 1, 0, "embed")
+      return urlArr.join("/")
+    } else if (ytIndex && embedModeIndex >= 0) {
+      // Url is already in the correct format
+      return url
+    } else {
+      // url is invalid (not youtube or bad format)
+      return false
+    }
+  }
+
+  const buildGalleryImages = images => {
+    return images.map(image => {
+      const src = image.fixed ? image.fixed.src : null
+      const title = image.title ? image.title : null
+      const caption = image.description ? image.description : null
+
+      return {
+        src,
+        title,
+        caption,
+      }
+    })
+  }
+
+  const handleOpenImage = e => {
+    e.preventDefault()
+    const src = e.target.src ? e.target.src : null
+
+    if (src) {
+      setShowLightbox(true)
+      setSelectedImage(src)
+    }
+  }
+
+  const handleCloseImage = e => {
+    e.preventDefault()
+    setShowLightbox(false)
+    setSelectedImage(null)
+  }
+
+  return (
+    <Layout active="single-resource" bgColor="magenta">
+      <SEO title="Resource" />
+      <HeaderArea
+        topLeft={returnHeaderLeft()}
+        topRight={width > 889 ? returnHeaderRight() : null}
+        customTop={true}
+      />
+      <div className="single-resource__container">
+        {showLightbox && selectedImage && (
+          <Dialog
+            className="single-resource__lightbox"
+            aria-label="image dialog"
+          >
+            <button onClick={handleCloseImage}>&#10006;</button>
+            <div
+              className="single-resource__lightbox-img"
+              style={{ backgroundImage: `url("${selectedImage}")` }}
+            ></div>
+          </Dialog>
+        )}
+        <SocialMediaBar title={title} url={url} hashtags={["arts-matter"]} />
+        {/* 
+              |     Move these sections to components later   |
+              V                                               V      */}
+        {instructionalResources && (
+          <InstructionalResources resources={instructionalResources} />
+        )}
+        {classroomPhotos && (
+          <div className="single-resource__classroom-photos">
+            <h4>Classroom Photos</h4>
+            <ImageGallery images={buildGalleryImages(classroomPhotos)} />
+          </div>
+        )}
+        {videos && (
```

Comment:
There are a few odd things happening in the video section.

**On mobile the student artwork section is overlaying the videos:**
<img width="754" alt="image" src="https://user-images.githubusercontent.com/22242017/108645176-94b8c780-7477-11eb-98f9-55e326b15c0b.png">

**The aspect ratio of the video is a little wonky, and should really be set explicitly.**

On mobile the videos are really tall, and on large screens they seem to wide. This may be as simple as setting the width and height parameters of the `iframe`. If not, we may want to look into leveraging a YouTube embed plugin for Gatsby to do some lifting for us. We can chat about this tomorrow.